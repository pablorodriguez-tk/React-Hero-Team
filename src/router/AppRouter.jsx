import React from 'react';
import { BrowserRouter as Router, Switch } from 'react-router-dom';
import { LoginScreen } from '../components/auth/LoginScreen';
import DashboardRoutes from './DashboardRoutes';
import PrivateRoute from './PrivateRoute';
import PublicRoute from './PublicRoute';

const AppRouter = () => {
  const auth = false; //TODO:crear estado para verificar autenticacion
  return (
    <Router>
      <div>
        <Switch>
          <PublicRoute
            exact
            path="/login"
            component={LoginScreen}
            isAuthenticated={auth} //TODO:crear estado para verificar autenticacion
          />
          <PrivateRoute
            path="/"
            component={DashboardRoutes}
            isAuthenticated={auth} //TODO:crear estado para verificar autenticacion
          />
        </Switch>
      </div>
    </Router>
  );
};

export default AppRouter;
