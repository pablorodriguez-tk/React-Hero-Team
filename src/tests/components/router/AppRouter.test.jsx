import React from 'react';
import '@testing-library/jest-dom/extend-expect';
import { render, waitFor } from '@testing-library/react';
import { Provider } from 'react-redux';
import thunk from 'redux-thunk';
import configureMockStore from 'redux-mock-store';
import AppRouter from '../../../router/AppRouter';
import { MemoryRouter } from 'react-router';

const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);

describe('Test on <AppRouter/>', () => {
  test('Should to show login screen if user is not authenticated', async () => {
    const initState = {
      auth: {
        isLoggedIn: false,
      },
      heroes: {
        heroTeam: [],
      },
      _persist: {
        version: -1,
        rehydrated: true,
      },
    };

    const store = mockStore(initState);

    const wrapper = render(
      <MemoryRouter>
        <Provider store={store}>
          <AppRouter />
        </Provider>
      </MemoryRouter>
    );

    const loginPageTitle = await wrapper.findByTestId('loginScreen');

    await waitFor(() => {
      expect(loginPageTitle).toBeInTheDocument();
    });

    expect(wrapper).toMatchSnapshot();
  });

  test('Should to show Hero screen if user is authenticated', async () => {
    const initState = {
      auth: {
        isLoggedIn: true,
        email: 'challenge@alkemy.org',
      },
      heroes: {
        heroTeam: [],
      },
      _persist: {
        version: -1,
        rehydrated: true,
      },
    };

    const store = mockStore(initState);

    const wrapper = render(
      <MemoryRouter>
        <Provider store={store}>
          <AppRouter />
        </Provider>
      </MemoryRouter>
    );

    const heroPageTitle = await wrapper.findByTestId('heroScreen');

    await waitFor(() => {
      expect(heroPageTitle).toBeInTheDocument();
    });
    expect(wrapper).toMatchSnapshot();
  });
});
