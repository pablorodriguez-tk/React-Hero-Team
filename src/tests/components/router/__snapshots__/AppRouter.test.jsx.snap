// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test on <AppRouter/> Should to show Hero screen if user is authenticated 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <nav
          class="navbar navbar-expand-sm navbar-dark bg-dark"
        >
          <div
            class="container"
          >
            <a
              class="navbar-brand"
              href="/heroteam"
            >
              Hero Team
            </a>
            <div
              class="navbar-collapse"
            >
              <div
                class="navbar-nav"
              >
                <a
                  class="nav-item nav-link"
                  href="/search"
                >
                  Search
                </a>
              </div>
            </div>
            <ul
              class="navbar-nav ml-auto"
            >
              <button
                class="nav-item nav-link btn"
              >
                Logout
              </button>
            </ul>
          </div>
        </nav>
        <div
          class="container mt-2"
        >
          <h1
            class="d-flex justify-content-center animate__animated animate__fadeIn"
            data-testid="heroScreen"
          >
            My hero team
          </h1>
          <div
            class="animate__animated animate__fadeIn"
          >
            <h2
              class="d-flex justify-content-center "
            >
              Type: 
              <b />
            </h2>
            <hr />
            <div
              class="row"
            >
              <div
                class="col-6 text-center"
              >
                <h2>
                  Average powerstats
                </h2>
                <ul
                  class="list-group list-group-flush"
                >
                  <li
                    class="list-group-item"
                  >
                    Combat: 
                    <b>
                      0.0
                    </b>
                  </li>
                  <li
                    class="list-group-item"
                  >
                    Durability: 
                    <b>
                      0.0
                    </b>
                  </li>
                  <li
                    class="list-group-item"
                  >
                    Intelligence: 
                    <b>
                      0.0
                    </b>
                  </li>
                  <li
                    class="list-group-item"
                  >
                    Power: 
                    <b>
                      0.0
                    </b>
                  </li>
                  <li
                    class="list-group-item"
                  >
                    Speed: 
                    <b>
                      0.0
                    </b>
                  </li>
                  <li
                    class="list-group-item"
                  >
                    Strength: 
                    <b>
                      0.0
                    </b>
                  </li>
                </ul>
              </div>
              <div
                class="col-6 text-center"
              >
                <h2>
                  Average height and weight
                </h2>
                <ul
                  class="list-group list-group-flush"
                >
                  <li
                    class="list-group-item"
                  >
                    Height: 
                    <b>
                      0.0
                       cm
                    </b>
                  </li>
                  <li
                    class="list-group-item"
                  >
                    Weight: 
                    <b>
                      0.0
                       Kg
                    </b>
                  </li>
                </ul>
              </div>
            </div>
          </div>
          <div
            class="animate__animated animate__fadeIn"
          />
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <nav
        class="navbar navbar-expand-sm navbar-dark bg-dark"
      >
        <div
          class="container"
        >
          <a
            class="navbar-brand"
            href="/heroteam"
          >
            Hero Team
          </a>
          <div
            class="navbar-collapse"
          >
            <div
              class="navbar-nav"
            >
              <a
                class="nav-item nav-link"
                href="/search"
              >
                Search
              </a>
            </div>
          </div>
          <ul
            class="navbar-nav ml-auto"
          >
            <button
              class="nav-item nav-link btn"
            >
              Logout
            </button>
          </ul>
        </div>
      </nav>
      <div
        class="container mt-2"
      >
        <h1
          class="d-flex justify-content-center animate__animated animate__fadeIn"
          data-testid="heroScreen"
        >
          My hero team
        </h1>
        <div
          class="animate__animated animate__fadeIn"
        >
          <h2
            class="d-flex justify-content-center "
          >
            Type: 
            <b />
          </h2>
          <hr />
          <div
            class="row"
          >
            <div
              class="col-6 text-center"
            >
              <h2>
                Average powerstats
              </h2>
              <ul
                class="list-group list-group-flush"
              >
                <li
                  class="list-group-item"
                >
                  Combat: 
                  <b>
                    0.0
                  </b>
                </li>
                <li
                  class="list-group-item"
                >
                  Durability: 
                  <b>
                    0.0
                  </b>
                </li>
                <li
                  class="list-group-item"
                >
                  Intelligence: 
                  <b>
                    0.0
                  </b>
                </li>
                <li
                  class="list-group-item"
                >
                  Power: 
                  <b>
                    0.0
                  </b>
                </li>
                <li
                  class="list-group-item"
                >
                  Speed: 
                  <b>
                    0.0
                  </b>
                </li>
                <li
                  class="list-group-item"
                >
                  Strength: 
                  <b>
                    0.0
                  </b>
                </li>
              </ul>
            </div>
            <div
              class="col-6 text-center"
            >
              <h2>
                Average height and weight
              </h2>
              <ul
                class="list-group list-group-flush"
              >
                <li
                  class="list-group-item"
                >
                  Height: 
                  <b>
                    0.0
                     cm
                  </b>
                </li>
                <li
                  class="list-group-item"
                >
                  Weight: 
                  <b>
                    0.0
                     Kg
                  </b>
                </li>
              </ul>
            </div>
          </div>
        </div>
        <div
          class="animate__animated animate__fadeIn"
        />
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Test on <AppRouter/> Should to show login screen if user is not authenticated 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <div
          class="container login-container animate__animated animate__fadeIn animate__faster "
        >
          <div
            class="col-md-12 login-form"
          >
            <h3
              data-testid="loginScreen"
            >
              Login
            </h3>
            <form>
              <div
                class="input-group mb-1"
              >
                <input
                  autocomplete="off"
                  class="form-control undefined"
                  id="email"
                  name="email"
                  placeholder="Email"
                  type="email"
                  value="challenge@alkemy.org"
                />
              </div>
              <div
                class="input-group mb-4"
              />
              <div
                class="input-group mb-1"
              >
                <input
                  class="form-control undefined"
                  id="password"
                  name="password"
                  placeholder="Password"
                  type="password"
                  value="react"
                />
              </div>
              <div
                class="input-group mb-5"
              />
              <button
                class="btnSubmit mb-3"
                type="submit"
              >
                Login
              </button>
            </form>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <div
        class="container login-container animate__animated animate__fadeIn animate__faster "
      >
        <div
          class="col-md-12 login-form"
        >
          <h3
            data-testid="loginScreen"
          >
            Login
          </h3>
          <form>
            <div
              class="input-group mb-1"
            >
              <input
                autocomplete="off"
                class="form-control undefined"
                id="email"
                name="email"
                placeholder="Email"
                type="email"
                value="challenge@alkemy.org"
              />
            </div>
            <div
              class="input-group mb-4"
            />
            <div
              class="input-group mb-1"
            >
              <input
                class="form-control undefined"
                id="password"
                name="password"
                placeholder="Password"
                type="password"
                value="react"
              />
            </div>
            <div
              class="input-group mb-5"
            />
            <button
              class="btnSubmit mb-3"
              type="submit"
            >
              Login
            </button>
          </form>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
